src/app/service/recap/ListItem.jsx

```diff
@@ -2,16 +2,29 @@ import React from 'react';
 import { Link } from 'react-router-dom';
 
 import Icon from '../../../components/icon';
+import { SERVICE_FIELDS } from '../../serviceForm/routes';
+import { DOCUMENT_FIELDS } from '../../documents/routes';
 
 import './ListItem.css';
 
+function filterRoutes(service, route){
+  return service.metadata[route.metaDataSection].find( metadata => metadata.field_name === route.fieldName)
+}
+
+function getProgress(service){
+  const totalNumberOfFields = SERVICE_FIELDS.length + DOCUMENT_FIELDS.length;
+  const f = filterRoutes.bind(this, service);
```

Comment:
What's the point of using an arrow function here? We are not binding the `this` object.

Reply:
I don’t think arrow function has an advantage here. Will discuss with you offline

Reply:
Well, you're defining an inline function and explicitly binding it to be lexically scoped. I'd argue that if that's not the place to use an arrow function, literally no place is.

Having said that, looking more at the code I don't even see why binding it is necessary. If scope is irrelevant, it would've been more straightforward to just curry.
And honestly, even that's probably overkill, since filterRoutes doesn't need to be a high-level function in the first place.

So, this could just be:

```
const hasMetadata = route => service.metadata[route.metaDataSection].some(metadata => metadata.field_name === route.fieldName);
const fieldsWithMetadata = SERVICE_FIELDS.filter(metadataFilter).length + DOCUMENT_FIELDS.filter(metadataFilter).length;
```

And that could be simplified further by concatting the arrays, so the whole thing could be just:

```
function getProgress(service) {
  const allFields = SERVICE_FIELDS.concat(DOCUMENT_FIELDS);

  const hasMetadata = route => service.metadata[route.metaDataSection].some(metadata => metadata.field_name === route.fieldName);
  const fieldsWithMetadata = allFields.filter(hasMetadata);

  return Math.round((fieldsWithMetadata.length / allFields.length) * 100)
}
```

But, to each his own I guess.